pkg/compiler/debug.go

```diff
@@ -0,0 +1,186 @@
+package compiler
+
+import (
+	"encoding/json"
+	"errors"
+	"go/ast"
+	"go/types"
+	"strconv"
+	"strings"
+)
+
+// DebugInfo represents smart-contract debug information.
+type DebugInfo struct {
+	EntryPoint string            `json:"entrypoint"`
+	Documents  []string          `json:"documents"`
+	Methods    []MethodDebugInfo `json:"methods"`
+	Events     []EventDebugInfo  `json:"events"`
+}
+
+// MethodDebugInfo represents smart-contract's method debug information.
+type MethodDebugInfo struct {
+	ID string `json:"id"`
+	// Name is the name of the method in a format "{namespace},{display-name}"
+	Name string `json:"name"`
+	// Range is the range of smart-contract's opcodes corresponding to the method.
+	Range debugRange `json:"range"`
+	// Parameters is a list of method's parameters.
+	Parameters []debugParam `json:"params"`
+	// ReturnType is method's return type.
+	ReturnType string   `json:"return-type"`
+	Variables  []string `json:"variables"`
+	// SeqPoints is a map between source lines and byte-code instruction offsets.
+	// It has following format: "{address}[{document-index}]{start-line}:{start-column}-{end-line}:{end-column}".
+	SeqPoints []string `json:"sequence-points"`
```

Comment:
This is a very complex thing packed into a `string`, why? It's OK for JSON to have it packed like that, but I'd rather have some more convenient representation here.